### PR TITLE
Support Builders for Native/JS

### PR DESCRIPTION
**Pull request by oldergod**
I'm not really happy with `jvmOnly` but that's what I came up with which would not introduce breaking changes.

### CODE REVIEW

File: wire-runtime/src/jsMain/kotlin/com/squareup/wire/Message.kt [hunk 1]
```diff
@@ -53,5 +54,73 @@ protected actual constructor(
   /**
    * Superclass for protocol buffer message builders.
    */
-  actual abstract class Builder<M : Message<M, B>, B : Builder<M, B>>
+  actual abstract class Builder<M : Message<M, B>, B : Builder<M, B>> protected constructor() {
+    /**
```
Comment (Egorand) [inline wire-runtime/src/jsMain/kotlin/com/squareup/wire/Message.kt:58]: These comments aren't inherited from the `expect Builder`, right?

File: wire-runtime/src/jsMain/kotlin/com/squareup/wire/Message.kt [hunk 2]
```diff
@@ -53,5 +54,73 @@ protected actual constructor(
...
+     * [Buffer] of the message's unknown fields that is lazily instantiated between calls to
+     * [buildUnknownFields]. It's automatically cleared in [buildUnknownFields], and can also be
+     * manually cleared by calling [clearUnknownFields].
+     */
+    internal var unknownFieldsBuffer: Buffer? = null
+    internal var unknownFieldsWriter: ProtoWriter? = null
+
+    fun addUnknownFields(unknownFields: ByteString): Builder<M, B> = apply {
+      if (unknownFields.size > 0) {
```
Comment (Egorand) [inline wire-runtime/src/jsMain/kotlin/com/squareup/wire/Message.kt:74]: There's probably no good way to extract this code to avoid duplication 😢

**Review by Egorand — APPROVED** [review]

File: wire-runtime/src/jsMain/kotlin/com/squareup/wire/Message.kt (hunk 1 above)
Comment (oldergod) [reply wire-runtime/src/jsMain/kotlin/com/squareup/wire/Message.kt:58]: But you're right, there are things I could move onto the `expect`

Comment (oldergod) [reply wire-runtime/src/jsMain/kotlin/com/squareup/wire/Message.kt]: Done!

**Comment by oldergod** [comment]
I could have otherwise created typealiases for `kotlin.jvm.*` annotations...

**Pull request merged**